lib/rust/ensogl/core/src/debug/stats.rs

```diff
@@ -210,6 +205,7 @@ impl StatsData {
         };
         self.frame_counter += 1;
```

Comment:
I was writing about it in another PR review – I do not understand why this field is here. Also, I think its name is very confusing. This is frame counter of only these frames where the stats were gathered, this is not "all frames counter", as the name may suggest.

Moreover, it seems that the only place this variable is being used is when it is compared to 0 to check if this is the first sample, so why we are collecting information of rendered frames if no one uses this information?

Reply:
**Re: `frame_counter`:** I changed the field to a boolean flag, thanks! Other than that, I'm not fully sure if what you were commenting on is just that it was a number, or that it exists at all? If the latter, then I don't currently see good alternatives:

 - I could add an `is_default()` method to `StatsData`, checking that every field equals zero, but that would be much more complex and also less effective;
 - I could add an `is_valid()` method checking e.g. just some fields (`self.fps != 0.0 || self.frame_counter != 0.0`), but I'm not convinced that would be more clear; an `initialized` field (or whatever name we'll agree upon) sounds most direct as to intention;
 - I can still name the field differently than `initialized`; however, e.g. `is_first_sample` doesn't sound appropriate to me, as this is notably not the *first* frame/sample that is "invalid", but the "0-th" - i.e., "the predecessor to the 1st frame" - and naming it `is_frame_zero` also sounds confusing to me, esp. given that programmers at various times count from 1 or from 0, so it's still not clear if 0 is "first" or not... other than that, I can't think of any other good names for it - I initially thought of `valid`, but I find it arguable which one is more clear.

Reply:
What about `Option<StatsData>`?

Reply:
@kazcw Initially (some time ago) I suggested to return `Option<StatsData>` from `begin_frame()`, but Wojciech then suggested to rather return `default()` than `None` in case of the "uninitialized" frame. Would your suggestion be to track this as `Option<StatsData>` inside, but then return `default()` from `begin_frame()` instead of `None`? Or to return `None` from `begin_frame()` in that case?